File: DSA-DAILY/DSA-DAILY/27-07-2025/2210. Count Hills and Valleys in an Array@leetcode-easy.py
class Solution:
    def countHillValley(self, nums: List[int]) -> int:
        res,j,n=0,0,len(nums)
        for i in range(1,n-1):
            if nums[j]<nums[i] and nums[i]>nums[i+1] or nums[j]>nums[i] and nums[i]<nums[i+1]:
                res+=1
                j=i

        return res

# Example usage
#    nums = [2, 4, 1, 1, 6, 5]
if __name__ =="__main__":
    sol=Solution()

    nums = list(map(int,input("Enter the array elements separated by space: ").split()))
    result=sol.countHillValley(nums)
    print("Number of hills and valleys:", result)

File: DSA-DAILY/DSA-DAILY/27-07-2025/2210. Count Hills and Valleys in an Array@leetcode-easy.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int countHillValley(vector<int>& nums) {
        int res=0,j=0;
        for(int i=1;i<nums.size()-1;i++){
            if((nums[i]>nums[j]&&nums[i]>nums[i+1])||(nums[i]<nums[j]&&nums[i]<nums[i+1])){
                res++;
                j=i;
            }
        }
        return res;
    }
};
int main() {
    Solution sol;
    vector<int> nums = {2, 4, 1, 1, 6, 5};
    int result = sol.countHillValley(nums);
    cout << "Number of hills and valleys: " << result << endl; // Output: 3
    return 0;
}
// Time Complexity: O(n)
// Space Complexity: O(1)